Modules/Core/Common/src/itkProcessObject.cxx

```diff
@@ -1115,16 +1095,7 @@ ProcessObject::DataObjectIdentifierType
 ProcessObject
 ::MakeNameFromIndex(DataObjectPointerArraySizeType idx) const
 {
-  if ( idx < ITK_GLOBAL_INDEX_NAMES_NUMBER )
-    {
-    return ProcessObject::DataObjectIdentifierType( globalIndexNames[idx] );
-    }
-  else
-    {
-    char buf[2+21]; // a 64-bit integer is ~20 decimal places max
-    sprintf(buf, "_%u", static_cast<unsigned int>(idx));
-    return buf;
-    }
+  return '_' + std::to_string(idx);
```

Comment:
How many dynamic memory allocations occur in this line? 2? 3?

Reply:
@blowekamp Good question! When `idx` is less than 100, I expect no dynamic allocation at all. Most common `std::string` implementations nowadays use small string optimization (SSO). 

I just had a closer look: With VS2017 x64, it appears that an `std::string` can hold up to 15 characters without any dynamic memory allocation! So with the proposed commit, `MakeNameFromIndex(idx)` would not do any dynamic memory allocation, as long as `idx` is less than `1.0e+14` (= 10^14). So the function would produce "_99999999999999", without dynamic allocation  `:-)`

Reply:
Update: I just checked that GCC and Clang also do small-string optimization for `std::string`, to avoid expensive dynamic memory allocation. So when `idx` has less than 15 decimal digits, I really don't think there will be any dynamic allocation, when doing `'_' + std::to_string(idx)`.

Note also that the current ITK git master version of `MakeNameFromIndex(idx)` has an integer overflow when `idx` has more than 10 decimal digits (specifically, when `idx > UINT_MAX`). The proposed commit will fix this bug, as it will support the full range of the type of `idx`.